### ISSUE TITLE
Support ClusterAccessConfig on EKS Cluster

### ISSUE THREAD

**Issue by blakeromano** [pulumi/pulumi-eks#1027]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

Want to be able to setup ClusterAccessConfig to adopt the new authentication method with users into K8s to move off the config map model.

### Affected area/feature

EKS authentication/Cluster Features

**Comment by mjeffryes** [comment]
Thanks for filing this request @blakeromano. If you have any other documentation links or details on the context/use case motivating your move off of the config map model that might be useful to who ever pick up this work.


AWS provider resource and example for AccessConfig [aws.eks.Cluster | Pulumi Registry](https://www.pulumi.com/registry/packages/aws/api-docs/eks/cluster/#eks-cluster-with-access-config)
CF property reference [AWS::EKS::Cluster AccessConfig - AWS CloudFormation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-eks-cluster-accessconfig.html)

**Comment by TarekAS** [comment]
Workaround using transformations:
```
const cluster = new eks.Cluster("cluster", {
    ...
}, {
    transformations: [args => {
        if (args.type === "aws:eks/cluster:Cluster") {
            args.props["accessConfig"] = {
                authenticationMode: "API"
            };
        }
        return undefined;
    }]
});
```

**Comment by flostadler** [comment]
This was released in version [v2.6.0](https://github.com/pulumi/pulumi-eks/releases/tag/v2.6.0)
